intersect_test.go

```diff
@@ -63,6 +64,39 @@ func TestHash(t *testing.T) {
 	assert.Equal(t, s, []interface{}{2})
 }
 
+func TestSortedGenericV2(t *testing.T) {
```

Comment:
Seems like two test cases - Please create a data struct which each test and then loop through them.